src/mono/mono/mini/interp/transform.c

```diff
@@ -8815,6 +8821,20 @@ interp_compute_native_offset_estimates (TransformData *td)
 			if (MINT_IS_EMIT_NOP (opcode))
 				continue;
 			noe += interp_get_ins_length (ins);
+
+			if (!final_code && td->optimized &&
+					(ins->flags & INTERP_INST_FLAG_CALL) &&
+					ins->info.call_info &&
+					ins->info.call_info->call_args) {
+				// When code is optimized, for a call, the offset allocator
+				// might end up inserting additional moves for the arguments
+				int *call_args = ins->info.call_info->call_args;
+				while (*call_args != -1) {
+					noe += 4; // mono_interp_oplen [MINT_MOV_VT];
```

Comment:
Is it always a single MINT_MOV_VT per argument?

Reply:
It is either a scalar mov (size 3) or a vt move (which also has size embedded so it is size 4). We are being conservative here since it is not worth checking the signature.